merge.go

```diff
@@ -451,12 +475,35 @@ func getArrayModifications(obj []interface{}, simpleList bool) []ModificationDef
 		case !isString:
 			//Do absolutely nothing
 
+		case mergeRegEx.MatchString(e): // check for (( merge ))
+			result = append(result, ModificationDefinition{listOp: listOpMergeOnKey})
+			continue
+
+		case mergeOnKeyRegEx.MatchString(e): // check for (( merge on "key" ))
+			/* #0 is the whole string,
+			 * #1 is string 'on'
+			 * #2 is the named-entry identifying key
+			 */
+			if captures := mergeOnKeyRegEx.FindStringSubmatch(e); len(captures) == 3 {
+				key := strings.TrimSpace(captures[2])
+				result = append(result, ModificationDefinition{listOp: listOpMergeOnKey, key: key})
+				continue
+			}
+
+		case inlineRegEx.MatchString(e): // check for (( inline ))
+			result = append(result, ModificationDefinition{listOp: listOpMergeInline})
+			continue
+
+		case replaceRegEx.MatchString(e): // check for (( replace ))
+			result = append(result, ModificationDefinition{listOp: listOpReplace})
+			continue
+
```

Comment:
Can you add some tests in  https://github.com/geofffranks/spruce/blob/master/merge_test.go#L184 to cover the new cases handled here, since the ShouldReplaceArray and ShouldInlineMergeArray functions and their tests were deleted?

Reply:
Sure, I checked the coverage report to verify the code portions are still somehow tested, but it makes perfect sense to explicitly test that. It kind of breaks the style not testing some of the operators that way.